gatewayFunctions.cmake

```diff
@@ -113,7 +113,7 @@ function(findAndInstall libraryName version submoduleRootDirectory cmakeRootDire
             #If the library directory doesn't exist, pull submodules
             if(NOT EXISTS "${cmakeRootDirectory}/CMakeLists.txt")
                 execute_process(
-                    COMMAND git submodule update --init ${submoduleRootDirectory}
+                    COMMAND git submodule update --init --recursive ${submoduleRootDirectory}
```

Comment:
We need to build the E2E tests to get the iothub_test library, and the E2E tests are pulling c files from c-shared.  This is probably an issue we need to resolve, so I just posted #323